### PR TITLE
Add table changes example, improve `TableChanges` documentation

### PR DESCRIPTION
**Pull request by OussamaSaoudi-db**
## What changes are proposed in this pull request?

This PR adds an example for using and reading table changes for a table through a CLI. 

We also introduce a cdf test on a table that was generated with various merge operations. This table contains both deletion vectors and cdc actions.

This PR also improves the documentation for the `table_changes` module so that example usage is clear in the user docs.

## How was this change tested?
Cargo doc and cargo test both run successfully

### CODE REVIEW

File: kernel/src/lib.rs [hunk 1]
```diff
@@ -10,12 +10,16 @@
...
+//! [read-table-multi-threaded] example). An example for reading the table changes for a table
+//! using the default engine can be found in the [read-table-changes] example.
+//!
 //!
 //! [read-table-single-threaded]:
 //! https://github.com/delta-io/delta-kernel-rs/tree/main/kernel/examples/read-table-single-threaded
 //! [read-table-multi-threaded]:
 //! https://github.com/delta-io/delta-kernel-rs/tree/main/kernel/examples/read-table-multi-threaded
+//![read-table-changes]:
+//! https://github.com/delta-io/delta-kernel-rs/tree/main/kernel/examples/read-table-changes
```
Comment (roeap) [inline kernel/src/lib.rs:22]:
```suggestion
//! [read-table-changes]:
//! https://github.com/delta-io/delta-kernel-rs/tree/main/kernel/examples/read-table-changes
```

**Review by roeap — APPROVED** [review]
LGTM.

File: kernel/examples/read-table-changes/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    let options = HashMap::from([("skip_signature", "true".to_string())]);
+    let engine = Arc::new(DefaultEngine::try_new(
+        table.location(),
+        options,
+        Arc::new(TokioBackgroundExecutor::new()),
+    )?);
+    let table_changes = table.table_changes(engine.as_ref(), cli.start_version, cli.end_version)?;
+
+    let x = table_changes.into_scan_builder().build()?;
+    let batches: Vec<RecordBatch> = x
```
Comment (zachschuermann) [inline kernel/examples/read-table-changes/src/main.rs:38]:
```suggestion
    let table_changes_scan = table_changes.into_scan_builder().build()?;
    let batches: Vec<RecordBatch> = table_changes_scan
```

File: kernel/src/table_changes/mod.rs [hunk 3]
```diff
@@ -1,4 +1,36 @@
...
+//! # use std::sync::Arc;
+//! # use delta_kernel::engine::sync::SyncEngine;
+//! # use delta_kernel::expressions::{column_expr, Scalar};
+//! # use delta_kernel::{Expression, Table, Error};
+//! # let path = "./tests/data/table-with-cdf";
+//! # let engine = Arc::new(SyncEngine::new());
+//! // Construct a table from a path oaeuhoanut
+//! let table = Table::try_from_uri(path)?;
+//!
+//! // Declare the version range for the table's change data feed
```
Comment (zachschuermann) [inline kernel/src/table_changes/mod.rs:14]:
```suggestion
//! // Get the table changes (change data feed) between version 0 and 1
```

File: kernel/src/table_changes/mod.rs [hunk 4]
```diff
@@ -1,4 +1,36 @@
...
+//! let table_changes = table.table_changes(engine.as_ref(), 0, 1)?;
+//!
+//! // Optionally specify a schema and predicate for the table changes scan
```
Comment (zachschuermann) [inline kernel/src/table_changes/mod.rs:17]:
```suggestion
//! // Optionally specify a schema and predicate to apply to the table changes scan
```

File: kernel/src/table_changes/mod.rs [hunk 5]
```diff
@@ -1,4 +1,36 @@
...
+//!
+//! // Construct the table changes scan
+//! let table_changes_scan = table_changes
+//!     .into_scan_builder()
+//!     .with_schema(schema)
+//!     .with_predicate(predicate.clone())
+//!     .build()?;
+//!
+//! // Execute the table changes scan to get a fallible iterator of `ScanResult`s
+//! let batches = table_changes_scan.execute(engine.clone())?;
```
Comment (zachschuermann) [inline kernel/src/table_changes/mod.rs:31]:
```suggestion
//! let table_change_batches = table_changes_scan.execute(engine.clone())?;
```

**Review by zachschuermann — APPROVED** [review]
looks good just nits

**Comment by zachschuermann** [comment]
removing `breaking-change` label - was failing on something in test_utils? but no breaking changes here

File: kernel/src/table_changes/mod.rs [hunk 6]
```diff
@@ -246,12 +250,7 @@ fn check_cdf_table_properties(table_properties: &TableProperties) -> DeltaResult
 }
 
 /// Ensures that Change Data Feed is supported for a table with this [`Protocol`] .
-//
-//  Currently the only read feature allowed is deletion vectors. This will be expanded in the
-//  future to support more delta table features.
-//
-//  Because only deletion vectors are supported, reader version 2 will not be allowed. That is
-//  because version 2 requires that column mapping is enabled. Reader versions 1 and 3 are allowed.
+//See the documentation of [`TableChanges`] for more details.
```
Comment (zachschuermann) [inline kernel/src/table_changes/mod.rs:253]:
```suggestion
// See the documentation of [`TableChanges`] for more details.
```

**Review by zachschuermann — APPROVED** [review]

**Pull request merged**